edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/Handlers/TwinHandler.cs

```diff
@@ -0,0 +1,350 @@
+            }
+            catch (Exception)
+            {
+                Events.MissingProxy(identity.Id);
+                return false;
+            }
+
+            var message = new EdgeMessage.Builder(publishInfo.Payload).Build();
+            await action(proxy, rid.Value);
+
+            return true;
+        }
+
+        Task StartProcessingLoop()
+        {
+            var loopTask = Task.Run(
+                                async () =>
+                                {
+                                    Events.ProcessingLoopStarted();
+
+                                    while (await this.notifications.Reader.WaitToReadAsync())
+                                    {
+                                        var processingInfo = await this.notifications.Reader.ReadAsync();
+
+                                        try
+                                        {
+                                            if (processingInfo.Action == ProcessingInfo.GetOrSet.Get)
+                                            {
+                                                await this.HandleTwinGet(processingInfo.Match, processingInfo.PublishInfo);
+                                            }
+                                            else
+                                            {
+                                                await this.HandleUpdateReported(processingInfo.Match, processingInfo.PublishInfo);
+                                            }
+                                        }
+                                        catch (Exception e)
+                                        {
+                                            Events.ErrorProcessingNotification(e);
+                                        }
+                                    }
+
+                                    Events.ProcessingLoopStopped();
+                                });
+
+            return loopTask;
+        }
+
+        static string GetTwinResultTopic(IIdentity identity, string statusCode, string correlationId)
+        {
+            var identityComponents = identity.Id.Split(HandlerUtils.IdentitySegmentSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            switch (identityComponents.Length)
+            {
+                case 1: return string.Format(TwinResultDevice, identityComponents[0], statusCode, correlationId);
+                case 2: return string.Format(TwinResultModule, identityComponents[0], identityComponents[1], statusCode, correlationId);
+
+                default:
+                    Events.BadIdentityFormat(identity.Id);
+                    throw new Exception($"cannot decode identity {identity.Id}");
+            }
+        }
+
+        static string GetDesiredPropertiesUpdateTopic(IIdentity identity, string version)
+        {
+            var identityComponents = identity.Id.Split(HandlerUtils.IdentitySegmentSeparator, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
if((identity is IModuleIdentity moduleIdentity)
{}
else if(identity is IDeviceIdentity deviceIdentity)
{
}
else
{
}

Reply:
ended up with switch:
switch (identity)
            {
                case IModuleIdentity moduleIdentity: ..
                case IDeviceIdentity deviceIdentity: ..
                default: